Frameworks/CoreText/CTLine.mm

```diff
@@ -16,16 +16,48 @@
 
 #import <CoreText/CTLine.h>
 #import <StubReturn.h>
-#import "NSStringInternal.h"
 #import "CoreTextInternal.h"
 #import "CGContextInternal.h"
 #import "DWriteWrapper_CoreText.h"
 #import <CoreText/CTTypesetter.h>
 
 #import <algorithm>
 #import <numeric>
+#import <cwchar>
 #import <vector>
 
+static CFStringRef __CTCreateReversedString(CFStringRef string) {
+    if (string == nullptr) {
+        return nullptr;
+    }
+
+    CFIndex length = CFStringGetLength(string);
+    CFIndex usedBufLen;
+    CFStringGetBytes(string, CFRangeMake(0, length), kCFStringEncodingUTF16, 0, false, nullptr, length, &usedBufLen);
+
+    if (length < 2) {
+        return CFStringCreateCopy(kCFAllocatorDefault, string);
+    }
+
+    CFIndex bufLen = usedBufLen + 1 * sizeof(UniChar);
+    woc::unique_iw<UniChar> characters(static_cast<UniChar*>(IwMalloc(bufLen)));
+    memset(characters.get(), '\0', bufLen);
+    CFStringGetBytes(
+        string, CFRangeMake(0, length), kCFStringEncodingUTF16, 0, false, reinterpret_cast<UInt8*>(characters.get()), usedBufLen, nullptr);
+
+    wchar_t* result = _wcsrev(reinterpret_cast<wchar_t*>(characters.get()));
+    if (result == nullptr) {
+        return nullptr;
+    }
+
+    return CFStringCreateWithBytesNoCopy(kCFAllocatorDefault,
+                                         reinterpret_cast<UInt8*>(characters.release()),
+                                         bufLen,
+                                         kCFStringEncodingUTF16,
+                                         false,
+                                         nullptr);
```

Comment:
you will need to provide an allocator to delete this string; look at the implementation of `initWithBytesNoCopy:freeWhenDone:` for NSString, expecting freeWhenDone=YES